Contact on different thirdparties with parent

leninrivas opened the pull request.

leninrivas committed:
Show contacts of parent

leninrivas committed:
Show contacs of parent in items

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -1666,6 +1666,15 @@ public function selectcontacts($socid, $selected = '', $htmlname = 'contactid',
+			$soc = new Societe($db);
+			$result = $soc->fetch($socid);
+			if ($result > 0 && !empty($soc->parent)) {
+				$sql .= ' OR sp.fk_soc = '.$soc->parent;
```
Surely a bug here. You add a OR be before you may have a AND and after another AND.
You must introduce parenthesis on test for fk_soc to avoid unexepected effects.
Also can you confirm that option CONTACT_ON_DIFFERENT_THIRDPARTIES means in fact
CONTACT_SHOW_ALSO_PARENT_COMPANY_CONTACTS (i mean it show also contact of parent company but no other thirdpary ?).
If yes can you use this name CONTACT_SHOW_ALSO_PARENT_COMPANY_CONTACTS

leninrivas replied:
Hi. Is right. Fixed

leninrivas committed:
Fixed to avoid unexepected effects

Fixed to avoid unexepected effects, change name options and add option with same parent.

leninrivas committed:
Change name option contact show also parent

leninrivas reviewed (commented):
Recomendations applied

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -1664,8 +1664,27 @@ public function selectcontacts($socid, $selected = '', $htmlname = 'contactid',
+				$sql .= ' OR sp.fk_soc IN (SELECT rowid FROM '.MAIN_DB_PREFIX.'societe WHERE parent = '.$soc->parent.')';
+			}
+    	}
+		$sql .= ')';
```
A parenthesis is closed in any case, but it is open only in one case. This is surely a bug.
Also can you check your IDE to set tabs to be tabs instead of spaces ? This generated trouble on merging and code analysis.

leninrivas replied:
I was testing an IDE

leninrivas committed:
Fix parenthesis bug

leninrivas committed:
Fix tab

leninrivas committed:
Fix tab

leninrivas committed:
Fix conflict    int $object

 $sql .= " WHERE t.fk_soc = ".((int) $object->id);

eldy committed:
Merge branch 'develop' into develop
